src/include/rbd_types.h

```diff
@@ -106,4 +107,9 @@ struct rbd_obj_header_ondisk {
 	struct rbd_obj_snap_ondisk snaps[0];
 } __attribute__((packed));
 
+enum group_image_link_state {
+	LINK_NORMAL,
```

Comment:
DETACHING is the same as ATTACHING. It only shows that the link is incomplete. If the link is incomplete you still can repeat addition as well as removal.